### DISCUSSION TITLE
SurvivalTree with min_samples_leaf based on observed events

### DISCUSSION THREAD

**Discussion by tomer1812** [sebp/scikit-survival#445] [Q&A] [answered]
Hi,

I'm attempting to prevent node splitting in a SurvivalTree when there are fewer than 6 observed events in either the right or left nodes after a split. 
Using min_samples_leaf didn't achieve the desired outcome as it accounts for all samples, not just the observed events. 
I also tried min_weight_fraction_leaf with the observed event indicator column as sample_weight, but this approach didn't work and resulted in a smaller number of observed events in each leaf. 
How can I properly configure the scikit-survival SurvivalTree to ensure a minimum number of observed events in each leaf?

Thank you!

**Comment by sebp** [comment, marked as answer]
Unfortunately, there is currently no way to achieve this. Also note that using `sample_weight` is currently broken (see #443).
